src/app.js

```diff
@@ -25,6 +25,36 @@ if ( !components ) {
 
 components.widget.toggle( true );
 components.widget.setState( 'pending' );
+components.api.onError( function ( errorCode ) {
+	components.widget.setState( 'err' );
+	components.widget.setErrorMessage( errorCode );
+} );
+components.api.onResult( function ( result ) {
+	// Insert modified HTML.
+	$( '.mw-parser-output' ).replaceWith( $( result.replacementHtml ) );
```

Comment:
`$( '.mw-parser-output' ).html( result.replacementHtml )` might be cleaner

Reply:
Done.